### PR TITLE
docs: reflect release of v9 in config migration guide

### PR DESCRIPTION
**Pull request by pwbriggs**
Update the eslintrc to flat config migration guide with the release of 9.0.0.

#### Prerequisites checklist

- [X] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[X] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)

I changed ["Configuration Migration Guide"](https://eslint.org/docs/latest/use/configure/migration-guide) to reflect that ESLint 9.0.0 has been released.

#### Is there anything you'd like reviewers to focus on?

Not sure of this is the best wording. It's unclear how this is different from usage in v8, as described in the following paragraph.

### CODE REVIEW

**Review by nzakas — APPROVED** [review]
LGTM. Just need to wait until start of next week to merge.

File: docs/src/use/configure/migration-guide.md
Comment (pwbriggs) [inline docs/src/use/configure/migration-guide.md:23]: @nzakas I'm not sure if this belongs in this PR, but it's unclear how to use flat config in v8&mdash;how is placing an `eslint.config.js` in the project root any different than v9 (does `.eslintrc.js` take precedence)? In what circumstances might you want to set `ESLINT_USE_FLAT_CONFIG`? If you're using flat config, wouldn't is be detected because of the `eslint.config.js` already present?

(I didn't edit that part of the docs because I'm not familiar with the specifics of configs.)

Comment (nzakas) [reply docs/src/use/configure/migration-guide.md:23]: I think those questions are already answered here:
https://eslint.org/docs/v8.x/use/configure/configuration-files-new

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks!

**Pull request merged**
